streams/jvm/src/main/scala/zio/stream/experimental/platform.scala

```diff
@@ -119,4 +125,288 @@ trait ZStreamPlatformSpecificConstructors {
+        } yield bytes
+      }
+    }
+
+  /**
+   * Creates a stream from the resource specified in `path`
+   */
+  final def fromResource(
+    path: String,
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[Any, IOException, Byte] =
+    ZStream.managed {
+      ZManaged.fromAutoCloseable {
+        ZIO.effectBlockingIO(getClass.getClassLoader.getResourceAsStream(path.replace('\\', '/'))).flatMap { x =>
+          if (x == null)
+            ZIO.fail(new FileNotFoundException(s"No such resource: '$path'"))
+          else
+            ZIO.succeedNow(x)
+        }
+      }
+    }.flatMap(is => fromInputStream(is, chunkSize = chunkSize))
+
+  /**
+   * Creates a stream from a `java.io.InputStream`. Ensures that the input
+   * stream is closed after it is exhausted.
+   */
+  def fromInputStreamEffect[R](
+    is: ZIO[R, IOException, InputStream],
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[R, IOException, Byte] =
+    fromInputStreamManaged(is.toManaged(is => ZIO.effectTotal(is.close())), chunkSize)
+
+  /**
+   * Creates a stream from a managed `java.io.InputStream` value.
+   */
+  def fromInputStreamManaged[R](
+    is: ZManaged[R, IOException, InputStream],
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[R, IOException, Byte] =
+    ZStream.managed(is).flatMap(fromInputStream(_, chunkSize))
+
+  /**
+   * Creates a stream from `java.io.Reader`.
+   */
+  def fromReader(reader: => Reader, chunkSize: Int = ZStream.DefaultChunkSize): ZStream[Any, IOException, Char] =
+    ZStream.fromEffect(UIO(reader)).flatMap { capturedReader =>
+      ZStream.repeatEffectChunkOption {
+        for {
+          bufArray  <- UIO(Array.ofDim[Char](chunkSize))
+          bytesRead <- ZIO.effectBlockingIO(capturedReader.read(bufArray)).asSomeError
+          chars <- if (bytesRead < 0)
+                     ZIO.fail(None)
+                   else if (bytesRead == 0)
+                     UIO(Chunk.empty)
+                   else if (bytesRead < chunkSize)
+                     UIO(Chunk.fromArray(bufArray).take(bytesRead))
+                   else
+                     UIO(Chunk.fromArray(bufArray))
+        } yield chars
+      }
+    }
+
+  /**
+   * Creates a stream from an effect producing `java.io.Reader`.
+   */
+  def fromReaderEffect[R](
+    reader: => ZIO[R, IOException, Reader],
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[R, IOException, Char] =
+    fromReaderManaged(reader.toManaged(r => ZIO.effectTotal(r.close())), chunkSize)
+
+  /**
+   * Creates a stream from managed `java.io.Reader`.
+   */
+  def fromReaderManaged[R](
+    reader: => ZManaged[R, IOException, Reader],
+    chunkSize: Int = ZStream.DefaultChunkSize
+  ): ZStream[R, IOException, Char] =
+    ZStream.managed(reader).flatMap(fromReader(_, chunkSize))
+
+  /**
+   * Creates a stream from a callback that writes to `java.io.OutputStream`.
+   * Note: the input stream will be closed after the `write` is done.
```

Comment:
This note is a bit misleading here, as the input stream is an implementation detail which is not visible on the API